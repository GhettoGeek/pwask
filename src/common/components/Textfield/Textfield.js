import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles'
import { TextField } from '@material-ui/core'

const useStyles = makeStyles((theme) => ({
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    width: 200,
  },
}))

function EnhancedTextField({
  id, label, variant, error, helperText,
}) {
  const classes = useStyles()
  const [value, setValue] = React.useState('')
  const handleChange = (event) => setValue(event.target.value)

  return (
    <TextField
      id={id}
      label={label}
      margin="normal"
      variant={variant}
      value={value}
      className={classes.textField}
      error={error}
      helperText={helperText}
      onChange={handleChange}
      inputProps={{
        'aria-label': label,
      }}
    />
  )
}

EnhancedTextField.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string,
  variant: PropTypes.string,
  error: PropTypes.bool,
  helperText: PropTypes.string,
}

EnhancedTextField.defaultProps = {
  id: 'standard-textfield',
  label: '',
  variant: 'standard',
  error: false,
  helperText: null,
}

export default EnhancedTextField
