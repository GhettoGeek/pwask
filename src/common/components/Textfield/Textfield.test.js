import React from 'react'
import { shallow } from 'enzyme'
import Textfield from './Textfield'

describe('Textfield', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(<Textfield label="Textfield" />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
