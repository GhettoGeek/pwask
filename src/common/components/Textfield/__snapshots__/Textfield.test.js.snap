// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield renders correctly the component 1`] = `"<div class=\\"MuiFormControl-root MuiTextField-root makeStyles-textField-1 MuiFormControl-marginNormal\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated\\" data-shrink=\\"false\\" for=\\"standard-textfield\\" id=\\"standard-textfield-label\\">Textfield</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"standard-textfield\\" value=\\"\\" aria-label=\\"Textfield\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div>"`;
